var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -0,0 +1,147 @@
+
+    # -
+    def __build_mods_coreneuron(self, mods_location, link_flag, include_flag):
+        spec = self.spec
+        assert os.path.isdir(mods_location), mods_location
+        include_flag += ' -I%s' % (spec['coreneuron'].prefix.include)
+        which('nrnivmodl-core')(
+            '-i', include_flag, '-l', link_flag, '-n', self.mech_name,
+            '-v', str(spec.version), '-c', mods_location)
+        output_dir = spec.architecture.target
+        mechlib = find_libraries("libcorenrnmech*", output_dir)
+        assert len(mechlib), "Error creating corenrnmech lib"
+        return mechlib
+
+    # -
+    def _get_lib_flags(self, lib_name):
+        """ Helper method to get linking flags similar to spack build, for solid deployments:
+              1. static libs passed via full path
+              2. shared libs passed with -L -l and RPATH flags
+            Attention: This func doesnt handle recursive deps of static libs.
+        """
+        spec = self.spec[lib_name]
+        if spec.satisfies('+shared'):  # Prefer shared if both exist
+            return " %s %s" % (spec.libs.rpath_flags, spec.libs.ld_flags)
+        return ' ' + spec.libs.joined()
+
+    # -
+    def install(self, spec, prefix, install_src=True):
+        """ Install:
+              bin/ <- special and special-core
+              lib/ <- hoc, mod and lib*mech*.so
+              share/ <- neuron & coreneuron mod.c's (modc and modc_core)
+        """
+        mkdirp(prefix.bin)
+        mkdirp(prefix.lib)
+        mkdirp(prefix.share.modc)
+
+        arch = spec.architecture.target
+        shutil.move(join_path(arch, 'special'), prefix.bin)
+        shutil.move(arch + "/.libs/libnrnmech.so", prefix.lib)
+        self._patch_special(prefix)
+
+        if spec.satisfies('+coreneuron'):
+            install = which('nrnivmech_install.sh', path=".")
+            install(prefix)
+
+        if install_src:
+            self._install_src(prefix)
+
+    @staticmethod
+    def _patch_special(prefix, libname='libnrnmech.so'):
+        which('sed')('-i',
+                     's#-dll .*#-dll %s "$@"#' % prefix.lib.join(libname),
+                     prefix.bin.special)
+
+    def _install_src(self, prefix):
+        """Copy original and translated c mods
+        """
+        arch = self.spec.architecture.target
+        shutil.move('mod', prefix.lib.mod)
+        shutil.move('hoc', prefix.lib.hoc)
+        shutil.move('common', prefix.lib.common)
+
+        for cmod in find(arch, "*.c", recursive=False):
+            shutil.move(cmod, prefix.share.modc)
+
+        if self.spec.satisfies('+coreneuron'):
+            shutil.move(join_path(arch, 'core/mod2c'), prefix.share.modc_core)
+
+    # -
+    def setup_environment(self, spack_env, run_env):
+        spack_env.unset('LC_ALL')
+        run_env.set('LIBNRNMECH_PATH', self.spec.prefix.lib.join('libnrnmech.so'))
+        run_env.set('BGLIBPY_MOD_LIBRARY_PATH', self.spec.prefix.lib.join('libnrnmech.so'))
```

Comment:
what's different between two? can't we just have one? for example `NRNMECH_LIB_PATH` ?

Reply:
I wrote in the PR. Temporarily, for compat with bglibpy, we should have BGLIBPY_MOD_LIBRARY_PATH. When they change to the new name we drop it.
I will change LIBNRNMECH_PATH to NRNMECH_LIB_PATH

Reply:
done